### PR TITLE
Bug 1127939 - Wait for provider ready before searching.

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[daleharvey Dale  Harvey (daleharvey)](https://github.com/daleharvey) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8b22974ad5c3)

File: apps/search/js/providers/suggestions.js [hunk 1]
```diff
@@ -37,13 +37,15 @@
...
     search: function(input) {
-      return new Promise((resolve, reject) => {
-        var url = encodeTerms(SearchProvider('suggestUrl'), input);
-        LazyLoader.getJSON(url, true).then(result => {
-          var results = result[1];
-          // We add an item to search the entered term as well
-          results.unshift(result[0]);
-          resolve(results);
+      return SearchProvider.ready().then(function() {
+        return new Promise((resolve, reject) => {
```
Comment (KevinGrandon) [inline apps/search/js/providers/suggestions.js]: Why do we need this nested promise? Can't we do all of this logic within the ```.then()```?

Comment (daleharvey) [inline apps/search/js/providers/suggestions.js]: We will need to resolve with the results?

Comment (KevinGrandon) [inline apps/search/js/providers/suggestions.js]: I think you can use the ```resolve``` as the first argument from the ```.then()```? So basically standard promise chaining..

```
return SearchProvider.ready().then(resolve => {

});
```

Comment (daleharvey) [inline apps/search/js/providers/suggestions.js]: ah nice I didnt realise that would work, will do

**Comment by try-server-hook** [comment]
[daleharvey Dale  Harvey (daleharvey)](https://github.com/daleharvey) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=6092992300c8)

File: apps/search/js/providers/suggestions.js (hunk 1 above)
Comment (daleharvey) [inline apps/search/js/providers/suggestions.js]: Changed it

Comment (daleharvey) [inline apps/search/js/providers/suggestions.js]: err no you cant do that, .then resolves with the value of the promise its called on

File: apps/search/js/providers/suggestions.js [hunk 2]
```diff
@@ -37,13 +37,15 @@
...
 
     search: function(input) {
-      return new Promise((resolve, reject) => {
-        var url = encodeTerms(SearchProvider('suggestUrl'), input);
-        LazyLoader.getJSON(url, true).then(result => {
-          var results = result[1];
-          // We add an item to search the entered term as well
-          results.unshift(result[0]);
-          resolve(results);
+      return SearchProvider.ready().then(resolve => {
```
Comment (KevinGrandon) [inline apps/search/js/providers/suggestions.js]: Ok, I think I mis-read the code before, sorry.  You're right, I think then() should just return a new promise as you're doing here.

This looks good, I think you should remove the resolve here then, that was wrong.

```js
return SearchProvider.ready().then(() => {
```

**Pull request closed without merging**